spec/requests/contact_embed_spec.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+RSpec.describe "Contact Embed" do
+  let(:edition) do
+    body_field = build(:field, id: "body", type: "govspeak")
+    document_type = build(:document_type, contents: [body_field])
+    create(:edition, document_type_id: document_type.id)
+  end
+
+  describe "GET /documents/:document/contact-embed" do
+    it "renders a service unavailable response when the Publishing API is down" do
+      stub_publishing_api_isnt_available
+      get contact_embed_path(edition.document)
+
+      expect(response).to have_http_status(:service_unavailable)
+      expect(response.body).to have_content(I18n.t!("contact_embed.new.api_down"))
+    end
+  end
+
+  describe "POST /documents/:document/contact-embed" do
+    let(:organisation) do
+      {
+        "content_id" => SecureRandom.uuid,
+        "internal_name" => "Organisation",
+      }
+    end
+
+    let(:contact) do
+      {
+        "content_id" => SecureRandom.uuid,
+        "title" => "Contact",
+        "links" => { "organisations" => [organisation["content_id"]] },
+      }
+    end
+
+    before do
+      stub_publishing_api_get_editions([contact], Contacts::EDITION_PARAMS)
+      stub_publishing_api_has_linkables([organisation],
+                                        document_type: "organisation")
+    end
+
+    it "returns just the embed code in a model context" do
+      post contact_embed_path(edition.document),
+           params: { contact_id: contact["content_id"] },
+           headers: { "Content-Publisher-Rendering-Context" => "modal" }
```

Comment:
Minor: if we're going to do this elsewhere, could we use a helper like `rendering_content("modal")`?

Reply:
I looked across all the request tests upcoming and it's only used in this and one other file so I don't think this is a big concern currently. I think we'd want to see it used a reasonable amount more to justify a helper. 

This is shaped somewhat by it not being the clearest helper to write. I guess we'd want it to return a hash and be passed into headers. For it to be global we'd probably want to name it with headers in the name, so it'd be used like: `headers: rendering_context_header("modal")` which didn't feel a huge gain for any ambiguities it creates.

Certainly happy to re-visit if usage proliferates and causes pain.